### DISCUSSION TITLE
Accessing strategy instance variables after backtest run

### DISCUSSION THREAD

**Discussion by zhang-alvin** [kernc/backtesting.py#294] [Q&A] [answered]
I'd like to access some of the indicators that I'm supplying for a backtest so that I can make custom plots. Using the tutorial example for reference:

```
from backtesting import Backtest, Strategy
from backtesting.lib import crossover

from backtesting.test import SMA, GOOG

#build strategy object based on abstract class
class SmaCross(Strategy):
    n1 = 10
    n2 = 20

    def init(self):
        close = self.data.Close
        self.sma1 = self.I(SMA, close, self.n1)
        self.sma2 = self.I(SMA, close, self.n2)

    def next(self):
        if crossover(self.sma1, self.sma2):
            self.buy()
        elif crossover(self.sma2, self.sma1):
            self.sell()

bt = Backtest(GOOG, SmaCross,
              cash=10000, commission=.002,
              exclusive_orders=True)

output = bt.run()
```
I'd like to be able to access the `SmaCross.sma1` and `SmaCross.sma2` array after the backtests completes. 
`bt._strategy` doesn't seem to be an instance of the strategy:

```
bt._strategy.sma1
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-6-9f7e77c45178> in <module>
----> 1 bt._strategy.sma1

AttributeError: type object 'SmaCross' has no attribute 'sma1'
```

Is there a simple way to access the indicators after the run? One workaround that can work is by making a copy of the indicators as class members (non-instance variables), but that seems messy. Any help would be greatly appreciated.

**Comment by kernc** [comment, marked as answer]
> `bt._strategy` doesn't seem to be an instance of the strategy

Where'd you get `bt._strategy`? It's [_`stats._strategy`_](https://kernc.github.io/backtesting.py/doc/examples/Quick%20Start%20User%20Guide.html#Optimization) (`stats = bt.run()`).

**Reply by zhang-alvin** [reply]
Ah of course, silly me. And to be consistent with the above example, it'd be `output._strategy.sma1`. Thanks @kernc.
